lib/gui/modules/analytics.js

```diff
@@ -120,17 +142,22 @@ analytics.service('AnalyticsService', function($log, $window, $mixpanel, Setting
    */
   this.logEvent = (message, data) => {
 
-    if (SettingsModel.get('errorReporting') && isRunningInAsar()) {
+    // Mixpanel expects a flat object
+    const flattenedData = data ? flatten(data, {
```

Comment:
We should extract this logic (and the start casing one) into a separate function and write unit tests for it.